BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -2854,6 +2863,7 @@ public void UpdateLaserSpecifics(bool newColor, bool newTex, bool newWidth, bool
                                 //Debug.Log("[BDArmory.ModuleWeapon]: barrelIndex for " + this.GetShortName() + " reset");
                             }
                         }
+                        if (animIndex + 1 > fireState.Length) animIndex = 0;
```

Comment:
`animIndex >= fireState.Length` is cleaner.

Reply:
yes, except it's dealing with a 0-based index, which means needing the +1 to convert to the 1-based firestate count; animIndex >= firestate.length would have it reset early.

Reply:
Sorry, but I don't follow your logic here. `a + 1 > b` is the same as `a >= b` for integers, which is what you'd use if `animIndex` was being used as an index into `fireState`.
If `animIndex` is supposed to go up to `fireState.Length` before resetting (i.e., `0` plus indices `1,...,fireState.Length`, where `animIndex = 0` doesn't correspond to an entry in `fireState`), then the condition ought to be `animIndex > fireState.Length`.

Reply:
Ok, sitting down and going through it again, yeah, you are correct. Guess I wasn't thinking straight when I answered you last time. Updated PR with change